### PR TITLE
[MU4] Accessibility for menu

### PR DESCRIPTION
**Pull request by Eism**
Resolves: 
#8370
#8373
#8375
#8376
#8377

### CODE REVIEW

File: src/appshell/qml/AppMenuBar.qml [hunk 1]
```diff
@@ -52,6 +53,8 @@ ListView {
     NavigationPanel {
         id: navPanel
         name: "AppMenuBar"
+        enabled: root.enabled && root.visible
+        accessible.name: qsTrc("appshell", "App menu bar") + " " + navPanel.directionInfo
```
Comment (shoogle) [inline src/appshell/qml/AppMenuBar.qml:57]: Including `navPanel.directionInfo` in the name is helpful at first but I think blind users will quickly get tired of hearing _"direction is horizontal"_ or _"direction is vertical"_ all the time. Instead, I think we should adopt the following convention:

- Toolbars and menu bars are always __horizontal__.
- Item views (e.g. tables and trees) are always __both directions__ unless there is only one column of data in the model.
- Everything else (e.g. menu items, lists, single-column tables and trees) are always __vertical__.

This should make it unnecessary to say the direction explicitly in most cases. If there are specific places where we need to break the convention then we can always say it in those places only.

@igorkorsukov, what do you think?

File: src/appshell/qml/AppTitleBar.qml [hunk 2]
```diff
@@ -82,10 +88,23 @@ Rectangle {
             anchors.right: parent.right
             anchors.verticalCenter: parent.verticalCenter
 
+            NavigationPanel {
+                id: navAppControlPanel
+                name: "AppControl"
+                section: navSec
+                order: 1
+                accessible.name: qsTrc("appshell", "App control") + " " + navAppControlPanel.directionInfo
```
Comment (shoogle) [inline src/appshell/qml/AppTitleBar.qml:96]: Same

**Review by shoogle — COMMENTED** [review]
Thanks for looking at this! It works for me on Windows (Narrator, not NVDA) and Linux (Orca).

Like all accessibility in MU4, the menus are not currently working on Windows with NVDA or macOS with VoiceOver. I have a few ideas for how we might fix that which I will try on another PR. It should be good to merge this one in the meantime.

Key navigation of the menus is working for me on all platforms; it's just the screen reader speech that works on some but not others.

File: src/appshell/qml/AppMenuBar.qml (hunk 1 above)
Comment (igorkorsukov) [reply src/appshell/qml/AppMenuBar.qml:57]: Yes, I agree with that.
During development, I added direction information for demonstration and debugging purposes only.
Now we need to think and solve all the moments as they should be in production.

**Comment by vshalkevich** [comment]
Works good!

File: src/appshell/qml/AppMenuBar.qml (hunk 1 above)
Comment (Eism) [reply src/appshell/qml/AppMenuBar.qml]: thanks, removed

**Review by shoogle — APPROVED** [review]

**Review by RomanPudashkin — APPROVED** [review]

**Comment by DmitryArefiev** [comment]
Checked #8370 #8373 #8375 #8376 #8377with Narrator on Win10 and Orca on Linux Mint 20.2 - FIXED

Accessibility doesn't work properly with NVDA (Win) and VoiceOver (Mac) - logged separately as #9155 , #9153
The issue with Orca logged as #9209

**Pull request merged**